src/kademlia/dht_storage.cpp

```diff
@@ -505,6 +505,20 @@ namespace
 			return m_counters;
 		}
 
+		dht_storage_items save_items(int const /*max_items*/) const override
+		{
+			dht_storage_items ret;
+
+			// TODO: implement here
+
+			return ret;
+		}
+
+		void load_items(dht_storage_items const& /*data*/) override
```

Comment:
I think you are right, but I need to check. That could solve the clear of the internal object in `session_impl`